test/unittests/execute_test.cpp

```diff
@@ -693,6 +694,12 @@ TEST(execute, memory_grow_custom_hard_limit)
         EXPECT_THAT(execute(*instance, 0, {input}), Result(expected));
     }
 
+    {
+        const auto instance_huge_hard_limit = instantiate(*module, {}, {}, {}, {}, 65536);
+        EXPECT_THAT(execute(*instance_huge_hard_limit, 0, {65536}), Result(-1));
```

Comment:
What is the goal here? According to the other test, 65536 total is acceptable. Why don't we add something like instantiate with 0 and enlarge to 65536.

Reply:
> What is the goal here? 

To test the check at high values of hard limit.

> According to the other test, 65536 total is acceptable. Why don't we add something like instantiate with 0 and enlarge to 65536.

It would allocate 4GB, but I think we want to keep unit tests light-weight.
(there is a spec test that does this)